aptos-move/aptos-vm/src/zkid_validation.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::move_vm_ext::{AptosMoveResolver, SessionExt};
+use anyhow::{Context, Result};
+use aptos_types::{
+    jwks::{jwk::JWK, PatchedJWKs},
+    on_chain_config::{CurrentTimeMicroseconds, OnChainConfig},
+    transaction::SignedTransaction,
+    vm_status::{StatusCode, VMStatus},
+    zkid::{ZkpOrOpenIdSig, MAX_ZK_ID_AUTHENTICATORS_ALLOWED},
+};
+use aptos_vm_logging::log_schema::AdapterLogSchema;
+use move_core_types::{language_storage::CORE_CODE_ADDRESS, move_resource::MoveStructType};
+
+fn get_current_time_onchain(resolver: &impl AptosMoveResolver) -> Result<CurrentTimeMicroseconds> {
+    let bytes = resolver
+        .get_resource(&CORE_CODE_ADDRESS, &CurrentTimeMicroseconds::struct_tag())?
+        .context("Failed to get CurrentTimeMicroseconds resource")?;
+    let onchain_timestamp_obj = bcs::from_bytes::<CurrentTimeMicroseconds>(&bytes)?;
+    Ok(onchain_timestamp_obj)
+}
+
+fn get_jwks_onchain(resolver: &impl AptosMoveResolver) -> Result<PatchedJWKs> {
+    let bytes = resolver
+        .get_resource(&CORE_CODE_ADDRESS, &PatchedJWKs::struct_tag())?
```

Comment:
Your error gets eaten by anyhow. If you want to return a result, you can simply do so by using `PartialVMResult`.

Reply:
This will be taken care of when we replace this with a Move function call, so no worries.